addons/web/controllers/main.py

```diff
@@ -701,13 +703,17 @@ def backup(self, master_pwd, name, backup_format = 'zip'):
 
     @http.route('/web/database/restore', type='http', auth="none", methods=['POST'], csrf=False)
     def restore(self, master_pwd, backup_file, name, copy=False):
+        temp_path = tempfile.mkstemp()[1]
```

Comment:
As per the doc: 
> Unlike TemporaryFile(), the user of mkstemp() is responsible for deleting the temporary file when done with it.
So it was mainly to allow the file deletion to be in the finally: block. I think it could've been done with NamedTempFile using some parameters but this way feels cleaner to me.